apbs/create-secret-and-configmap-during-provision.md

```diff
@@ -0,0 +1,44 @@
+# Abstract
+
+This document outlines a proposal to create an additional configmap for services accessible via a mobile client, this configmap would compliment the secret we currently create during the provision of each service via an APB 
+
+## Terms
+
+- Mobile Client (a mobile device running a developers application)
+- APB (ansible playbook bundle)
+
+## Problem Description
+
+Our current approach is to create a secret during the provision of a new mobile service. This secret contains all of the information required to allow the UI to show the service and how to access the service, it also includes any information needed to create the mobile client configuration. This means we expose too much information and also need to filter out any sensitive information (usernames password etc) when constructing our mobile configuration to avoid it being exposed to the client and ending up in the public domain. As each service is different, it often requires a custom piece of code to filter out the fields that we don't want or need exposed. 
+
+## Proposed Solution
+
+When provisioning a new service instance, we should create two configuration resources:
+
+1) A secret containing private information such as usernames and passwords or anything that we do not want to be exposed to the mobile client.
+2) A configmap containing all of the required information needed for a mobile client to use the service.
```

Comment:
for push, the auth for device registration (done via HTTP_BASIC) would be needed in that configmap - since the client has to know that variantID:secret (aka user/password)

Reply:
yeah I think if it is needed by the client then it should go in the configmap